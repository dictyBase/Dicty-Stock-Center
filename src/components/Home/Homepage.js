// @flow
import React from "react"
import { connect } from "react-redux"
import { Helmet } from "react-helmet"
import Grid from "@material-ui/core/Grid"
import bowser from "bowser"
import Availability from "./Availability"
import OtherMaterials from "./OtherMaterials"
import Slideshow from "./Slideshow"
import Intro from "./Intro"
import About from "./About"
import BrowserWarning from "./BrowserWarning"
import HomepageColumn from "./HomepageColumn"
import LinkList from "./LinkList"
import StandardOperatingProcedures from "./StandardOperatingProcedures"
import { AuthenticatedUser } from "utils/apiClasses"
import {
  downloadLinks,
  infoLinks,
  materialsLinks,
  miscLinks,
} from "constants/linkLists"
import useStyles from "./homeStyles"

type Props = {
  /** the User object from the current state */
  user: Object,
  /** the user's first and last names, taken from the AuthenticatedUser class */
  fullName: string,
}

/**
 * Homepage is the main homepage component for DSC.
 */

const Homepage = ({ fullName, user }: Props) => {
  const classes = useStyles()

  return (
    <div className={classes.container}>
      <Helmet>
        <title>Dicty Stock Center</title>
        <meta
          name="description"
          content="The Dicty Stock Center is a rapidly growing central repository for Dictyostelium discoideum strains and those of related species, plasmids, commonly used food bacteria, and other materials such as antibodies."
        />
      </Helmet>
      {user && (
        <span>
          <h3>Hello, {`${fullName}!`}</h3>
        </span>
      )}
      {bowser.msie && bowser.version <= 10 && <BrowserWarning />}
      <Grid container justify="space-between" spacing={3}>
        <Grid item>
          <h1 className={classes.header}>
            Welcome to Dicty Stock Center (DSC)
          </h1>
        </Grid>
        <Grid item xs={12}>
          <Intro />
        </Grid>
        <HomepageColumn
          components={[<About />, <LinkList list={miscLinks} bgColor="blue" />]}
        />
        <HomepageColumn
          components={[
            <LinkList list={infoLinks} bgColor="blue" />,
            <Availability />,
            <OtherMaterials />,
            <StandardOperatingProcedures />,
          ]}
        />
        <HomepageColumn
          components={[
            <Slideshow />,
            <LinkList list={materialsLinks} bgColor="gray" />,
            <LinkList list={downloadLinks} bgColor="gray" />,
          ]}
        />
      </Grid>
    </div>
  )
}

const mapStateToProps = state => {
  if (state.auth.user) {
    const userData = new AuthenticatedUser(state.auth.user)
    return {
      user: state.auth.user,
      fullName: userData.getFullName(),
    }
  }
  return {}
}

export { Homepage }
export default connect<*, *, *, *, *, *>(mapStateToProps)(Homepage)
