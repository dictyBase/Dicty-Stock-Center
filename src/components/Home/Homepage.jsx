// @flow
import React, { Component } from "react"
import { connect } from "react-redux"
import { Flex, Box } from "rebass"
import bowser from "bowser"
import Links from "./Links"
import Info from "./Info"
import Availability from "./Availability"
import OtherMaterials from "./OtherMaterials"
import Downloads from "./Downloads"
import Slideshow from "./Slideshow"
import Materials from "./Materials"
import Intro from "./Intro"
import About from "./About"
import BrowserWarning from "./BrowserWarning"
import StandardOperatingProcedures from "./StandardOperatingProcedures"
import { AuthenticatedUser } from "utils/apiClasses"
import { HomepageHeader, Container } from "styles"
import type { MapStateToProps } from "react-redux"

type Props = {
  /** the User object from the current state */
  user: Object,
  /** the user's first and last names, taken from the AuthenticatedUser class */
  fullName: string
}

/**
 * This is the main homepage component for DSC
 */

class Homepage extends Component<Props> {
  renderGreeting = () => {
    return (
      <span>
        <h3>Hello, {`${this.props.fullName}!`}</h3>
      </span>
    )
  }
  render() {
    return (
      <Container>
        {this.props.user && this.renderGreeting()}
        {bowser.msie && bowser.version <= 10 && <BrowserWarning />}
        <Flex wrap justify="space-between">
          <Box>
            <HomepageHeader>
              <h1>Welcome to Dicty Stock Center (DSC)</h1>
            </HomepageHeader>
          </Box>
          <Box w={1}>
            <Intro />
          </Box>
          <Box w={[1, "30%", "30%"]}>
            <Flex column>
              <Box>
                <About />
              </Box>
              <Box>
                <Links />
              </Box>
            </Flex>
          </Box>
          <Box w={[1, "30%", "30%"]}>
            <Flex column>
              <Box>
                <Info />
              </Box>
              <Box>
                <Availability />
              </Box>
              <Box>
                <OtherMaterials />
              </Box>
              <Box>
                <StandardOperatingProcedures />
              </Box>
            </Flex>
          </Box>
          <Box w={[1, "33%", "33%"]}>
            <Flex justify="center" column>
              <Box>
                <Slideshow />
              </Box>
              <Box>
                <Materials />
              </Box>
              <Box>
                <Downloads />
              </Box>
            </Flex>
          </Box>
        </Flex>
      </Container>
    )
  }
}

const mapStateToProps: MapStateToProps<*, *, *> = state => {
  if (state.auth.user) {
    const userData = new AuthenticatedUser(state.auth.user)
    return {
      user: state.auth.user,
      fullName: userData.getFullName()
    }
  }
  return {}
}

export default connect(mapStateToProps)(Homepage)
