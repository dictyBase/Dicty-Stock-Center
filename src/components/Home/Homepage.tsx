import React from "react"
import { Helmet } from "react-helmet"
import Grid from "@material-ui/core/Grid"
import bowser from "bowser"
import Availability from "./Availability"
import OtherMaterials from "./OtherMaterials"
import Slideshow from "./Slideshow"
import Intro from "./Intro"
import About from "./About"
import BrowserWarning from "./BrowserWarning"
import HomepageColumn from "./HomepageColumn"
import LinkList from "./LinkList"
import StandardOperatingProcedures from "./StandardOperatingProcedures"
import { useAuthStore } from "components/authentication/AuthStore"
import {
  downloadLinks,
  infoLinks,
  materialsLinks,
  miscLinks,
} from "constants/linkLists"
import useStyles from "./homeStyles"

/**
 * Homepage is the main homepage component for DSC.
 */

const Homepage = () => {
  const classes = useStyles()
  const [{ user }] = useAuthStore()

  const fullName = `${user.first_name} ${user.last_name}`

  return (
    <div className={classes.container}>
      <Helmet>
        <title>Dicty Stock Center</title>
        <meta
          name="description"
          content="The Dicty Stock Center is a rapidly growing central repository for Dictyostelium discoideum strains and those of related species, plasmids, commonly used food bacteria, and other materials such as antibodies."
        />
      </Helmet>
      {user.data && (
        <span>
          <h3>Hello, {`${fullName}!`}</h3>
        </span>
      )}
      {bowser.msie && bowser.version <= 10 && <BrowserWarning />}
      <Grid container justify="space-between" spacing={3}>
        <Grid item>
          <h1 className={classes.header}>
            Welcome to Dicty Stock Center (DSC)
          </h1>
        </Grid>
        <Grid item xs={12}>
          <Intro />
        </Grid>
        <HomepageColumn
          components={[<About />, <LinkList list={miscLinks} bgColor="blue" />]}
        />
        <HomepageColumn
          components={[
            <LinkList list={infoLinks} bgColor="blue" />,
            <Availability />,
            <OtherMaterials />,
            <StandardOperatingProcedures />,
          ]}
        />
        <HomepageColumn
          components={[
            <Slideshow />,
            <LinkList list={materialsLinks} bgColor="gray" />,
            <LinkList list={downloadLinks} bgColor="gray" />,
          ]}
        />
      </Grid>
    </div>
  )
}

export default Homepage
