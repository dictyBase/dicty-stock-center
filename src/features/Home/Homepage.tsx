import React from "react"
import { Helmet } from "react-helmet"
import Grid from "@material-ui/core/Grid"
import Typography from "@material-ui/core/Typography"
import Availability from "./Availability"
import OtherMaterials from "./OtherMaterials"
import Slideshow from "./Slideshow"
import EditablePanel from "./EditablePanel"
import BrowserWarning from "./BrowserWarning"
import HomepageColumn from "./HomepageColumn"
import LinkList from "./LinkList"
import StandardOperatingProcedures from "./StandardOperatingProcedures"
import { useAuthStore } from "features/Authentication/AuthStore"
import useSupportedBrowsers from "common/hooks/useSupportedBrowsers"
import {
  downloadLinks,
  infoLinks,
  materialsLinks,
  miscLinks,
} from "common/constants/linkLists"
import useStyles from "./homeStyles"

const metaDesc =
  "The Dicty Stock Center is a rapidly growing central repository for Dictyostelium discoideum strains and those of related species, plasmids, commonly used food bacteria, and other materials such as antibodies."

/**
 * Homepage is the main homepage component for DSC.
 */

const Homepage = () => {
  const classes = useStyles()
  const [{ isAuthenticated, user }] = useAuthStore()
  const { supportedBrowser } = useSupportedBrowsers()
  const fullName = `${user.first_name} ${user.last_name}`

  return (
    <div>
      <Helmet>
        <title>Dicty Stock Center</title>
        <meta name="description" content={metaDesc} />
      </Helmet>
      {isAuthenticated && (
        <span>
          <h3>Hello, {`${fullName}!`}</h3>
        </span>
      )}
      {!supportedBrowser && <BrowserWarning />}
      <Grid container justify="space-between" spacing={3}>
        <Grid item className={classes.header}>
          <Typography variant="h1">
            Welcome to Dicty Stock Center (DSC)
          </Typography>
        </Grid>
        <Grid item xs={12} className={classes.intro}>
          <EditablePanel slug="dsc-intro" skeletonCount={5} />
        </Grid>
        <HomepageColumn
          components={[
            <EditablePanel slug="dsc-about" skeletonCount={10} />,
            <LinkList list={miscLinks} bgColor="blue" />,
          ]}
        />
        <HomepageColumn
          components={[
            <LinkList list={infoLinks} bgColor="blue" />,
            <Availability />,
            <OtherMaterials />,
            <StandardOperatingProcedures />,
          ]}
        />
        <HomepageColumn
          components={[
            <Slideshow />,
            <LinkList list={materialsLinks} bgColor="gray" />,
            <LinkList list={downloadLinks} bgColor="gray" />,
          ]}
        />
      </Grid>
    </div>
  )
}

export default Homepage
