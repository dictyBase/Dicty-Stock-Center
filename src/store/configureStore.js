import { createStore, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'
import rootReducer from '../reducers'
import DevTools from '../containers/DevTools'

const enhancer = compose(
    applyMiddleware(thunk),
    DevTools.instrument(),
)

export default function configureStore(initialState) {
  // todo -> create store with middleware

    const store = createStore(rootReducer, initialState, enhancer)

    if (module.hot) {
        module.hot.accept('../reducers', () =>
          store.replaceReducer(require('../reducers').default)
        )
    }
    return store
}
