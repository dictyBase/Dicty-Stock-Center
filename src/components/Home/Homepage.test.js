import React from "react"
import { Provider } from "react-redux"
import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import { Homepage } from "./Homepage"
import configureStore from "store"

const store = configureStore()

test("matching a snapshot of Home/Homepage", () => {
  const props = {
    user: {},
    fullName: "Jane Doe",
  }

  const component = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Homepage {...props} />
      </BrowserRouter>
    </Provider>,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
