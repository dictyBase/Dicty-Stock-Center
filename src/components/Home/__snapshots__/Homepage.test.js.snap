// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Home/Homepage 1`] = `
.c1 h1 {
  font-size: 48px;
  margin-top: 0px;
  margin-bottom: 25px;
  border-bottom: 1px solid #eee;
}

.c0 {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

.c4 {
  background-color: #f2f2f2;
  border: 1px solid #f2f2f2;
  margin: 15px auto;
  padding: 15px;
  border-radius: 10px;
  width: 100%;
}

.c2 {
  background-color: #eff8fb;
  border: 1px solid #eff8fb;
  margin: 15px auto;
  padding: 15px;
  border-radius: 10px;
  width: 100%;
}

.c3 ul,
.c3 li {
  margin: 0;
  padding: 0;
  list-style: none;
}

.c3 a {
  font-size: 18px;
}

.c5 {
  color: #961818;
}

@media (min-width:768px) {
  .c0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .c0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .c0 {
    width: 1170px;
  }
}

<div
  className="c0"
>
  <span>
    <h3>
      Hello, 
      Jane Doe!
    </h3>
  </span>
  <div
    className="sc-htpNat ftFunm sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="iHZvIS"
      is={null}
    >
      <div
        className="c1"
      >
        <h1>
          Welcome to Dicty Stock Center (DSC)
        </h1>
      </div>
    </div>
    <div
      className="gRrvFh"
      is={null}
    >
      <div
        className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
        is={null}
      >
        <div
          className="eJiGdf"
          is={null}
        >
          <div
            className="react-loading-theme-0"
          >
            <span>
              <span
                className="react-loading-skeleton"
                style={
                  Object {
                    "animation": "progress 1.2s ease-in-out infinite",
                  }
                }
              >
                ‌
              </span>
              <span
                className="react-loading-skeleton"
                style={
                  Object {
                    "animation": "progress 1.2s ease-in-out infinite",
                  }
                }
              >
                ‌
              </span>
              <span
                className="react-loading-skeleton"
                style={
                  Object {
                    "animation": "progress 1.2s ease-in-out infinite",
                  }
                }
              >
                ‌
              </span>
              <span
                className="react-loading-skeleton"
                style={
                  Object {
                    "animation": "progress 1.2s ease-in-out infinite",
                  }
                }
              >
                ‌
              </span>
              <span
                className="react-loading-skeleton"
                style={
                  Object {
                    "animation": "progress 1.2s ease-in-out infinite",
                  }
                }
              >
                ‌
              </span>
            </span>
            <br />
            <br />
          </div>
        </div>
      </div>
    </div>
    <div
      className="lmjRPw"
      is={null}
    >
      <div
        className="sc-htpNat pyeRb sc-bdVaJa iHZvIS"
        is={null}
      >
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
            is={null}
          >
            <div
              className="FHiEA"
              is={null}
            >
              <div
                className="react-loading-theme-1"
              >
                <span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                </span>
                <br />
                <br />
                <span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className="c2"
          >
            <div
              className="c3"
            >
              <ul>
                <li>
                  <a
                    href="/contact"
                    onClick={[Function]}
                  >
                    Contact the DSC
                  </a>
                </li>
                <li>
                  <a
                    href="/information/faq"
                    onClick={[Function]}
                  >
                    DSC FAQ
                  </a>
                </li>
                <li>
                  <a
                    href="/information/nomenclature-guidelines"
                  >
                    Nomenclature Guide
                  </a>
                </li>
                <li>
                  <a
                    href="/information/other-stock-centers"
                    onClick={[Function]}
                  >
                    Other Stock Centers
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="lmjRPw"
      is={null}
    >
      <div
        className="sc-htpNat pyeRb sc-bdVaJa iHZvIS"
        is={null}
      >
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className="c2"
          >
            <div
              className="c3"
            >
              <ul>
                <li>
                  <a
                    href="/information/order"
                    onClick={[Function]}
                  >
                    Order Information
                  </a>
                </li>
                <li>
                  <a
                    href="/information/payment"
                    onClick={[Function]}
                  >
                    Payment Information
                  </a>
                </li>
                <li>
                  <a
                    href="/information/deposit"
                    onClick={[Function]}
                  >
                    Deposit Information
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className="c4"
          >
            <h4>
              Strain & Plasmid Availability
            </h4>
            <h5>
              <strong>
                1927
              </strong>
               
              Strains
            </h5>
            <h5>
              <strong>
                882
              </strong>
               
              Plasmids
            </h5>
          </div>
        </div>
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
            is={null}
          >
            <div
              className="FHiEA"
              is={null}
            >
              <div
                className="react-loading-theme-2"
              >
                <span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                  <span
                    className="react-loading-skeleton"
                    style={
                      Object {
                        "animation": "progress 1.2s ease-in-out infinite",
                      }
                    }
                  >
                    ‌
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className="c4"
          >
            <a
              className="c5"
              href="https://northwestern.box.com/s/p0f8m70whgiuib2u0wt8gtn497ncmq8i"
              rel="noopener noreferrer"
              target="_blank"
            >
              Standard Operating Procedures
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="gFCdqg"
      is={null}
    >
      <div
        className="sc-htpNat bPnPne sc-bdVaJa iHZvIS"
        is={null}
      >
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className={undefined}
          >
            <div
              className="carousel carousel-slider"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <button
                className="control-arrow control-prev"
                onClick={[Function]}
                type="button"
              />
              <div
                className="slider-wrapper axis-horizontal"
                style={Object {}}
              >
                <ul
                  className="slider animated"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "MozTransform": "translate3d(0%,0,0)",
                      "MozTransitionDuration": "350ms",
                      "MsTransform": "translate3d(0%,0,0)",
                      "MsTransitionDuration": "350ms",
                      "OTransform": "translate3d(0%,0,0)",
                      "OTransitionDuration": "350ms",
                      "WebkitTransform": "translate3d(0%,0,0)",
                      "WebkitTransitionDuration": "350ms",
                      "msTransform": "translate3d(0%,0,0)",
                      "msTransitionDuration": "350ms",
                      "transform": "translate3d(0%,0,0)",
                      "transitionDuration": "350ms",
                    }
                  }
                >
                  <li
                    className="slide selected"
                    onClick={[Function]}
                  >
                    <div>
                      <img
                        alt="ctr9-mutant-DG1071"
                        src="ctr9-mutant-DG1071.png"
                      />
                    </div>
                  </li>
                  <li
                    className="slide"
                    onClick={[Function]}
                  >
                    <div>
                      <img
                        alt="gbqA-mutant-DG1120"
                        src="gbqA-mutant-DG1120.png"
                      />
                    </div>
                  </li>
                  <li
                    className="slide"
                    onClick={[Function]}
                  >
                    <div>
                      <img
                        alt="ggtA-mutant-DG1109"
                        src="ggtA-mutant-DG1109.png"
                      />
                    </div>
                  </li>
                  <li
                    className="slide"
                    onClick={[Function]}
                  >
                    <div>
                      <img
                        alt="tipB-mutant-DG1036"
                        src="tipB-mutant-DG1036.png"
                      />
                    </div>
                  </li>
                </ul>
              </div>
              <button
                className="control-arrow control-next"
                onClick={[Function]}
                type="button"
              />
            </div>
          </div>
        </div>
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className="c4"
          >
            <div
              className="c3"
            >
              <ul>
                <li>
                  <a
                    href="/strains"
                    onClick={[Function]}
                  >
                    Strain Catalog
                  </a>
                </li>
                <li>
                  <a
                    href="/plasmids"
                    onClick={[Function]}
                  >
                    Plasmid Catalog
                  </a>
                </li>
                <li>
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Bacterial Strains
                  </a>
                </li>
                <li>
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Other Materials
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="iHZvIS"
          is={null}
        >
          <div
            className="c2"
          >
            <h3>
              Download / View
            </h3>
            <div
              className="c3"
            >
              <ul>
                <li>
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Phenotype Ontology
                  </a>
                </li>
                <li>
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Strain Characteristics
                  </a>
                </li>
                <li>
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Mutagenesis Methods
                  </a>
                </li>
                <li>
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Plasmid Keywords
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
